===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ PortObjectAddPort:n$3 == { -1 } }) free(*(PortObjectAddPort:poi)) at 3 (line 810, column 4) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/snort-2.9.13/src/sfutil/sfportobject.c
+++ b/table1/benchmarks/snort-2.9.13/src/sfutil/sfportobject.c
@@ -807,7 +807,10 @@ int PortObjectAddPort( PortObject * po, int port, int not_flag )
    poi->lport = (unsigned short)port;
    poi->hport = (unsigned short)port;
 
-   return  sflist_add_tail( po->item_list, poi );
+   tmp = sflist_add_tail( po->item_list, poi );
+   if (tmp == -1)
+      free(poi);
+   return tmp;
 }
 
 /*
